nuxeo-platform-video-core/src/main/java/org/nuxeo/ecm/platform/video/tools/operations/ExtractClosedCaptionsFromVideo.java

```diff
@@ -0,0 +1,100 @@
+ *     Thibaud Arguillere
+ *     Ricardo Dias
+ */
+package org.nuxeo.ecm.platform.video.tools.operations;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.collectors.BlobCollector;
+import org.nuxeo.ecm.automation.core.util.BlobList;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.impl.blob.FileBlob;
+import org.nuxeo.ecm.platform.video.tools.VideoToolsService;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Extracts Closed Captions from a Video.
+ * 
+ * @since 8.4
+ */
+@Operation(id = ExtractClosedCaptionsFromVideo.ID, category = Constants.CAT_CONVERSION, label = "Extracts closed captions from the video.", description = "Extracts the closed captions from the whole video or from a part of it when startAt and end time is provided. The output format references how the output is generated, and xpath can be used to indicate the video blob when using documents.", aliases = {
+        "Video.ExtractClosedCaptions" })
+public class ExtractClosedCaptionsFromVideo {
+
+    public static final String ID = "Video.ExtractClosedCaptions";
+
+    @Param(name = "outFormat", required = false, values = { "txt" })
+    protected String outFormat;
+
+    @Param(name = "startAt", required = false)
+    protected String startAt;
+
+    @Param(name = "endAt", required = false)
+    protected String endAt;
+
+    @Param(name = "xpath", required = false, values = { "file:content" })
+    protected String xpath;
+
+    @OperationMethod
+    public Blob run(DocumentModel input) throws OperationException {
+        String blobPath = (!StringUtils.isEmpty(xpath)) ? xpath : "file:content";
+        return run((Blob) input.getPropertyValue(blobPath));
+    }
+
+    @OperationMethod
+    public BlobList run(DocumentModelList input) throws OperationException {
+        BlobList blobList = new BlobList();
+        for (DocumentModel doc : input) {
+            blobList.add((run(doc)));
+        }
+        return blobList;
+    }
+
+    @OperationMethod(collector = BlobCollector.class)
+    public Blob run(Blob input) throws OperationException {
+        try {
+            VideoToolsService service = Framework.getService(VideoToolsService.class);
+            Blob result = service.extractClosedCaptions(input, outFormat, startAt, endAt);
+            if (result == null) {
+                File tempFile = Framework.createTempFile("NxVT-", "txt");
+                tempFile.deleteOnExit();
+                Framework.trackFile(tempFile, this);
```

Comment:
`Framework.trackFile` is enough. Don't ever call `deleteOnExit`. See https://jira.nuxeo.com/browse/NXP-7276 for instance. I don't see what this has to do with empty files.